src/taxonomy/service.ts

```diff
@@ -0,0 +1,38 @@
+import { User } from '../auth'
+import { Taxonomy } from './taxonomy'
+
+export interface Job {}
+
+export interface TaxonomyService {
+  // ==== CREATE
+  create(options: { user: User; surveyId: string; taxonomy: Taxonomy }): Promise<Taxonomy>
+
+  // ==== READ
+  count(options: { surveyId: string }): Promise<number>
+
+  getMany(options: {
+    draft: boolean
+    limit: number
+    offset: number
+    search: string
+    surveyId: string
+    validate: boolean
+  }): Promise<Array<Taxonomy>>
+
+  get(options: { surveyId: string; taxonomyUuid: string; draft: boolean; validate: boolean }): Promise<Taxonomy>
+
+  // ==== UPDATE
+
+  updateTaxonomyProp(options: {
+    user: User
+    surveyId: string
+    taxonomyUuid: string
+    key: string
+    value: any
+  }): Promise<void>
```

Comment:
call this update and pass only props of type TaxonomyProps. not key value anymore